### ISSUE TITLE
Please complete and merge ekeys/fullscreen in master or replace it

### ISSUE THREAD

**Issue by RalfWerner** [termux/termux-x11#54]
@twaik has encouraged me to open an issue when there are still problems with X11. I did that in #38 (ToDo was now postponed to Discussions) with 10 Commits, which were there _off-topic_.
The X11 development of recent months has led to various versions that have been insignificantly/potentially improved compared to the **base** of September 2019 and is partially deteriorated. The biggest potential has 1.2.04 by @suhan-paradkar  and @agnostic-apollo. Here advantages and disadvantages of three versions (If github `*.apk` versions are expired, use [my](https://www.dropbox.com/s/ug071qoox8gwf1c/ip.zip?dl=0) copies):

- [1.0](https://www.dropbox.com/s/n85m6zkxkcvrk9p/dev-x11.apk?dl=0) **basic** the only one with fullscreen mode but the 8 local `ekeys` only together with OS-KB usable and no Reseed-Screen.
- [1.2.04](https://github.com/termux/termux-x11/actions/runs/1277414726) has the `ekeys` from/like Termux and Reseed-Screen. Unfortunately, the key-states and fullscreen are missing/lost but several other problems such as `PASTE` are solved - no progress in last weeks and fullscreen restauration failed.
- [1.2.06](https://github.com/termux/termux-x11/actions/runs/2076074339) has the `ekeys` from **base** (1.0) but permanent and Reseed-Screen and is independent of the Termux installation. This version, in addition to the advantage of independent signing, also has many disadvantages, which is why, I do not recommend the use see [OTS addendum](https://github.com/termux/termux-float/issues/37#issuecomment-916777123).

The Reseed-screen-choice _should_ adjust the X11 screen to the [residual surface](https://github.com/termux/termux-x11/issues/38#issuecomment-929064401) after deduction of EK/OS-KB and considering [fullscreen mode](https://github.com/termux/termux-x11/issues/29). The EK-lines are faulty in all versions. This refers to the positioning (overlapped) and the Reseed Screen. @agnostic-apollo  has made many changes in Termux to get an acceptable EK version. A common EK would probably be the simplest solution when [fullscreen mode](https://github.com/termux/termux-app/pull/1923#issuecomment-918252727) allowed in both apps.

I made checks/tests with different findings on three real devices and five AVD. If you are interested in, say it. I return to **base** again!

**Comment by RalfWerner** [comment]
@suhan-paradkar The [PR 19](https://github.com/termux/termux-x11/pull/19#issuecomment-998576293) has not been edited since last year and the last adjustments (`master` merge trial) have only made the start more complicated. _**Is there a schedule to continue or complete?**_

**Comment by Grimler91** [comment]
@RalfWerner don't ask for time estimates. This is an open source project, no one is getting paid to work on it and there are hence no deadlines or schedules.

If the feature is important for your use-case then consider trying to code it yourself and open a PR (WIP or finished) for feedback

**Comment by suhan-paradkar** [comment]
Closing since #19 is closed

**Comment by twaik** [comment]
@RalfWerner try latest build from [here](https://github.com/twaik/termux-x11/actions).

**Comment by RalfWerner** [comment]
Sorry @twaik I just came to check today. I did that with the last artifact and the one before it, but unfortunately had no success with both.
`-rw-rw---- 1 u0_a259 media_rw 22140512 Mar  3 11:59 i/app-debug.apk`
`-rw-rw---- 1 u0_a259 media_rw 22181472 Mar  4 12:18 sd/Pictures/app-debug.apk`
First three Fotos from **new** and **old=basic** of two mobiles (left and right)
![grafik](https://user-images.githubusercontent.com/45426380/222900079-c1084fd3-b850-4562-b282-9712cecfb0c1.png)
The left is my new Samsung A33 with Android 13 and right my old Huawei phone (Android 9) with SSM effects when power cable is off and *basic* version. 
- Foto after start of Xserver and Xclient (here my **check.c** with fullscreen). New: The processes of XWayland and check are running (last Foto), Old: use _real_ size of screen inclusive nav-bars and top-message-line.
- Preferences New: by App-Screen - no Notification and no Xclient like right. Direct mode is missing. Old: by Notification without nav-bar
- Termux with nav-bar and top-message-line and _*my*_ ExtraKey-Settings. In *1.2.04* Version (not installed on this device), this would be the same in X11. In both you find stdout on *check*.

my [OTS](https://github.com/termux/termux-float/issues/37#issuecomment-916777123) script and process check is here:
```
alias pp='ps -eo pid,ppid,comm,etime'; export XDG_RUNTIME_DIR=$TMPDIR; termux-x11 :0&
pp; DISPLAY=:0 TMP/check.exe i/PEK.xpm&
pp
# 
[~]$ pp
  PID  PPID COMMAND             ELAPSED
 6676  6674 bash               03:41:16
19838   960 com.termux         05:22:25
27447  6676 main                  00:33
27468 27447 Xwayland              00:32
27489  6676 check.exe             00:07
```
**_Was fehlt um weiter zu checken?_**

**Comment by twaik** [comment]
Does it actually work? I see "Not connected" screen.

**Comment by RalfWerner** [comment]
PID: 27468 and 27489 above are running! What is missing to connected?

**Comment by twaik** [comment]
I am not sure. I need logs to check what is actually happening. Invoke `export TERMUX_X11_DEBUG=1` before starting `termux-x11`.

**Comment by RalfWerner** [comment]
<details><summary> log </summary>

```
[~]$ pp
  PID  PPID COMMAND             ELAPSED
18261 19992 sshd               02:20:14
19838   960 com.termux         05:56:56
19894 19838 bash               05:56:55
19992     1 sshd               05:56:53
29312 19992 sshd                  13:57
29327 29312 bash                  13:54
29352   960 com.termux.api        13:53
30173 29327 main                  00:19
30193 30173 Xwayland              00:19
30222 29327 check.exe             00:06
30224 29327 ps                    00:01
[~]$ Somebody connected!
New client connection!
Not using O_TMPFILE
▒ Mg>
<   >     <2,k=1067556119,Ret:6/6/0 l=1
WxH:1080x2320 n=82/32, d=0/2
com.termux.x11 logcat: --------- beginning of crash
com.termux.x11 logcat: 03-04 13:57:17.127 23748 23782 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xc in tid 23782 (Thread-3), pid 23748 (com.termux.x11)
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : Build fingerprint: 'samsung/a33xnseea/a33x:13/TP1A.220624.014/A336BXXS4BWA1:user/release-keys'
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : Revision: '4'
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : ABI: 'arm64'
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : Processor: '2'
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : Timestamp: 2023-03-04 13:57:17.215528503+0100
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : Process uptime: 4644s
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : Cmdline: com.termux.x11
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : pid: 23748, tid: 23782, name: Thread-3  >>> com.termux.x11 <<<
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : uid: 10348
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
com.termux.x11 logcat: 03-04 13:57:17.774 27271 27271 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x000000000000000c
com.termux.x11 logcat: --------- beginning of main
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]: performTraversals params={(0,0)(fillxfill) sim={adjust=resize forwardNavigation} ty=BASE_APPLICATION wanim=0x1030309
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fl=81810580
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   pfl=1a020040
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   vsysui=1706
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   bhv=SHOW_TRANSIENT_BARS_BY_SWIPE
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fitSides= naviIconColor=0
com.termux.x11 logcat: 03-04 14:34:36.350 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   sfl=100000}
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]: performTraversals mFirst=true windowShouldResize=true viewVisibilityChanged=false mForceNextWindowRelayout=false params={(0,0)(fillxfill) sim={adjust=resize forwardNavigation} ty=BASE_APPLICATION fmt=TRANSLUCENT wanim=0x1030309
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fl=81810580
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   pfl=1a020040
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   vsysui=1706
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   bhv=SHOW_TRANSIENT_BARS_BY_SWIPE
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fitSides= naviIconColor=0
com.termux.x11 logcat: 03-04 14:34:36.351 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   sfl=100000}
com.termux.x11 logcat: 03-04 14:34:36.359 30227 30235 I com.termux.x11: Compiler allocated 4177KB to compile void android.widget.TextView.<init>(android.content.Context, android.util.AttributeSet, int, int)
com.termux.x11 logcat: 03-04 14:34:36.365 30227 30227 D BLASTBufferQueue: new BLASTBufferQueue, mNativeObject= 0x6ecd28bc30 name= ViewRootImpl@8fce065[MainActivity] sc.mNativeObject= 0x6e5d294370 caller= android.view.ViewRootImpl.updateBlastSurfaceIfNeeded:2789 android.view.ViewRootImpl.relayoutWindow:9635 android.view.ViewRootImpl.performTraversals:3759 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665 android.view.Choreographer$CallbackRecord.run:1301 android.view.Choreographer$CallbackRecord.run:1309 android.view.Choreographer.doCallbacks:923 android.view.Choreographer.doFrame:852 android.view.Choreographer$FrameDisplayEventReceiver.run:1283
com.termux.x11 logcat: 03-04 14:34:36.365 30227 30227 D BLASTBufferQueue: update, w= 1080 h= 2320 mNativeObject= 0x6ecd28bc30 sc.mNativeObject= 0x6e5d294370 format= -3 caller= android.graphics.BLASTBufferQueue.<init>:76 android.view.ViewRootImpl.updateBlastSurfaceIfNeeded:2789 android.view.ViewRootImpl.relayoutWindow:9635 android.view.ViewRootImpl.performTraversals:3759 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665
com.termux.x11 logcat: 03-04 14:34:36.367 30227 30227 I ViewRootImpl@8fce065[MainActivity]: Relayout returned: old=(0,80,1080,2400) new=(0,80,1080,2400) req=(1080,2320)0 dur=13 res=0x3 s={true 0x6f5d2b19d0} ch=true seqId=0
com.termux.x11 logcat: 03-04 14:34:36.369 30227 30227 D ViewRootImpl@8fce065[MainActivity]: mThreadedRenderer.initialize() mSurface={isValid=true 0x6f5d2b19d0} hwInitialized=true
com.termux.x11 logcat: 03-04 14:34:36.376 30227 30227 D SurfaceView@7dd0040: updateSurface: has no frame
com.termux.x11 logcat: 03-04 14:34:36.376 30227 30227 D SurfaceView@accb79: updateSurface: has no frame
com.termux.x11 logcat: 03-04 14:34:36.379 30227 30247 I ViewRootImpl@8fce065[MainActivity]: Resizing android.view.ViewRootImpl@952eb60: frame = [0,80][1080,2400] reportDraw = true forceLayout = false syncSeqId = 0
com.termux.x11 logcat: 03-04 14:34:36.380 30227 30253 D OpenGLRenderer: eglCreateWindowSurface
com.termux.x11 logcat: 03-04 14:34:36.385 30227 30227 I SurfaceView@7dd0040: windowStopped(false) true android.view.SurfaceView{7dd0040 VFE...... .F....ID 0,0-1080,2320 #7f0900f9 app:id/lorieView} of ViewRootImpl@8fce065[MainActivity]
com.termux.x11 logcat: 03-04 14:34:36.385 30227 30227 D SurfaceView@7dd0040: updateSurface: has no frame
com.termux.x11 logcat: 03-04 14:34:36.385 30227 30227 I SurfaceView@accb79: windowStopped(false) false android.view.SurfaceView{accb79 I.E...... ......I. 0,0-1,1 #7f09008c app:id/cursorView} of ViewRootImpl@8fce065[MainActivity]
com.termux.x11 logcat: 03-04 14:34:36.385 30227 30227 D SurfaceView@accb79: updateSurface: has no frame
com.termux.x11 logcat: 03-04 14:34:36.386 30227 30227 D ViewRootImpl@8fce065[MainActivity]: reportNextDraw android.view.ViewRootImpl.performTraversals:4313 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665 android.view.Choreographer$CallbackRecord.run:1301 android.view.Choreographer$CallbackRecord.run:1309
com.termux.x11 logcat: 03-04 14:34:36.387 30227 30227 I SurfaceView@7dd0040: Changes: creating=true format=true size=true visible=true alpha=false hint=false mUseAlpha=false visible=true left=true top=true
com.termux.x11 logcat: 03-04 14:34:36.390 30227 30227 D BLASTBufferQueue: update, w= 1080 h= 2320 mNativeObject= 0x6ecd28a330 sc.mNativeObject= 0x6e5d2b3b60 format= 4 caller= android.view.SurfaceView.createBlastSurfaceControls:1384 android.view.SurfaceView.updateSurface:1080 android.view.SurfaceView.lambda$new$0$android-view-SurfaceView:200 android.view.SurfaceView$$ExternalSyntheticLambda5.onPreDraw:2 android.view.ViewTreeObserver.dispatchOnPreDraw:1121 android.view.ViewRootImpl.performTraversals:4317
com.termux.x11 logcat: 03-04 14:34:36.390 30227 30227 I SurfaceView@7dd0040: 131924032 Cur surface: Surface(name=null)/@0x5d8eb23
com.termux.x11 logcat: 03-04 14:34:36.390 30227 30227 I SurfaceView@7dd0040: pST: sr = Rect(0, 0 - 1080, 2320) sw = 1080 sh = 2320
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 D SurfaceView@7dd0040: performSurfaceTransaction RenderWorker position = [0, 0, 1080, 2320] surfaceSize = 1080x2320
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: applyTransactionOnVriDraw: vri = ViewRootImpl@8fce065[MainActivity] fRS = false t = 0x6e8d28a570 android.view.SurfaceView.performSurfaceTransaction:959 android.view.SurfaceView.updateSurface:1106 android.view.SurfaceView.lambda$new$0$android-view-SurfaceView:200
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I ViewRootImpl@8fce065[MainActivity]: applyTransactionOnDraw: mRemoved = false isHardwareEnabled = true t = 0x6e8d28a570android.view.SurfaceView.applyTransactionOnVriDraw:2049 android.view.SurfaceView.performSurfaceTransaction:959 android.view.SurfaceView.updateSurface:1106 android.view.SurfaceView.lambda$new$0$android-view-SurfaceView:200 android.view.SurfaceView$$ExternalSyntheticLambda5.onPreDraw:2
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: updateSurface: mVisible = true mSurface.isValid() = true
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: updateSurface: mSurfaceCreated = false surfaceChanged = true visibleChanged = true
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: visibleChanged -- surfaceCreated
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: surfaceCreated 1 #8 android.view.SurfaceView{7dd0040 VFE...... .F....ID 0,0-1080,2320 #7f0900f9 app:id/lorieView}
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: surfaceChanged -- format=4 w=1080 h=2320
com.termux.x11 logcat: 03-04 14:34:36.391 30227 30227 I SurfaceView@7dd0040: surfaceChanged (1080,2320) 1 #8 android.view.SurfaceView{7dd0040 VFE...... .F....ID 0,0-1080,2320 #7f0900f9 app:id/lorieView}
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30227 E LorieX11Client: Surface: got new surface 0x6f5d2ba220
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client: Surface: changing surface 0x0 to 0x6f5d2ba220
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30227 I SurfaceView@7dd0040: surfaceRedrawNeeded
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client: Refreshing screen failed: Error getting shm image through MIT-SHM extension
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client: XCB Error of failed request:               Shm::BadSeg
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client:   Major opcode of failed request:          130 (Shm)
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client:   Minor opcode of failed request:          4 (GetImage)
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client:   Serial number of failed request:         143
com.termux.x11 logcat: 03-04 14:34:36.392 30227 30257 E LorieX11Client:   Current serial number in output stream:  143
com.termux.x11 logcat: 03-04 14:34:36.393 30227 30227 V SurfaceView@7dd0040: Layout: x=0 y=0 w=1080 h=2320, frame=Rect(0, 0 - 1080, 2320)
com.termux.x11 logcat: 03-04 14:34:36.393 30227 30227 I SurfaceView@accb79: Changes: creating=false format=true size=true visible=false alpha=false hint=false mUseAlpha=false visible=false left=true top=true
com.termux.x11 logcat: 03-04 14:34:36.394 30227 30227 D ViewRootImpl@8fce065[MainActivity]: Setup new sync id=0
com.termux.x11 logcat: 03-04 14:34:36.394 30227 30227 D ViewRootImpl@8fce065[MainActivity]: Setting syncFrameCallback
com.termux.x11 logcat: 03-04 14:34:36.394 30227 30227 E SurfaceSyncer: Failed to find sync for id=0
com.termux.x11 logcat: 03-04 14:34:36.394 30227 30227 D ViewRootImpl@8fce065[MainActivity]: registerCallbacksForSync syncBuffer=false
com.termux.x11 logcat: 03-04 14:34:36.403 30227 30276 D SurfaceView@7dd0040: updateSurfacePosition RenderWorker, frameNr = 1, position = [0, 0, 1080, 2320] surfaceSize = 1080x2320
com.termux.x11 logcat: 03-04 14:34:36.403 30227 30275 I ViewRootImpl@8fce065[MainActivity]: mWNT: t=0x6e8d28a570 mBlastBufferQueue=0x6ecd28bc30 fn= 1 caller= android.view.ViewRootImpl.lambda$applyTransactionOnDraw$11$android-view-ViewRootImpl:12629 android.view.ViewRootImpl$$ExternalSyntheticLambda1.onFrameDraw:4 android.graphics.HardwareRenderer$FrameDrawingCallback.onFrameDraw:924
com.termux.x11 logcat: 03-04 14:34:36.404 30227 30276 I SurfaceView@7dd0040: uSP: rtp = Rect(0, 0 - 1080, 2320) rtsw = 1080 rtsh = 2320
com.termux.x11 logcat: 03-04 14:34:36.404 30227 30276 I SurfaceView@7dd0040: onSSPAndSRT: pl = 0 pt = 0 sx = 1.0 sy = 1.0
com.termux.x11 logcat: 03-04 14:34:36.404 30227 30275 D BLASTBufferQueue: mergeWithNextTransaction, mNativeObject= 0x6ecd28bc30 t.mNativeObject= 0x6e8d28a570 frameNumber= 1 caller= android.view.ViewRootImpl.mergeWithNextTransaction:12594 android.view.ViewRootImpl.lambda$applyTransactionOnDraw$11$android-view-ViewRootImpl:12629 android.view.ViewRootImpl$$ExternalSyntheticLambda1.onFrameDraw:4 android.graphics.HardwareRenderer$FrameDrawingCallback.onFrameDraw:924 android.view.ViewRootImpl$6.onFrameDraw:2053 android.view.ThreadedRenderer$1.onFrameDraw:777
com.termux.x11 logcat: 03-04 14:34:36.404 30227 30275 D ViewRootImpl@8fce065[MainActivity]: Received frameDrawingCallback syncResult=0 frameNum=1.
com.termux.x11 logcat: 03-04 14:34:36.404 30227 30275 D ViewRootImpl@8fce065[MainActivity]: Setting up sync and frameCommitCallback
com.termux.x11 logcat: 03-04 14:34:36.404 30227 30276 I SurfaceView@7dd0040: aOrMT: ViewRootImpl@8fce065[MainActivity] t = android.view.SurfaceControl$Transaction@e53ab38 fN = 1 android.view.SurfaceView.-$$Nest$mapplyOrMergeTransaction:0 android.view.SurfaceView$SurfaceViewPositionUpdateListener.positionChanged:1538 android.graphics.RenderNode$CompositePositionUpdateListener.positionChanged:373
com.termux.x11 logcat: 03-04 14:34:36.405 30227 30276 I ViewRootImpl@8fce065[MainActivity]: mWNT: t=0x6e8d28c070 mBlastBufferQueue=0x6ecd28bc30 fn= 1 caller= android.view.SurfaceView.applyOrMergeTransaction:1456 android.view.SurfaceView.-$$Nest$mapplyOrMergeTransaction:0 android.view.SurfaceView$SurfaceViewPositionUpdateListener.positionChanged:1538
com.termux.x11 logcat: 03-04 14:34:36.405 30227 30276 D BLASTBufferQueue: mergeWithNextTransaction, mNativeObject= 0x6ecd28bc30 t.mNativeObject= 0x6e8d28c070 frameNumber= 1 caller= android.view.ViewRootImpl.mergeWithNextTransaction:12594 android.view.SurfaceView.applyOrMergeTransaction:1456 android.view.SurfaceView.-$$Nest$mapplyOrMergeTransaction:0 android.view.SurfaceView$SurfaceViewPositionUpdateListener.positionChanged:1538 android.graphics.RenderNode$CompositePositionUpdateListener.positionChanged:373 android.graphics.RenderNode$PositionUpdateListener.callPositionChanged:286
com.termux.x11 logcat: 03-04 14:34:36.434 30227 30253 D BLASTBufferQueue: [ViewRootImpl@8fce065[MainActivity]#0](f:0,a:0) onFrameAvailable the first frame is available
com.termux.x11 logcat: 03-04 14:34:36.435 30227 30253 D ViewRootImpl@8fce065[MainActivity]: Received frameCommittedCallback lastAttemptedDrawFrameNum=1 didProduceBuffer=true
com.termux.x11 logcat: 03-04 14:34:36.435 30227 30253 D OpenGLRenderer: CFMS:: SetUp Pid : 30227    Tid : 30253
com.termux.x11 logcat: 03-04 14:34:36.437 30227 30253 W Parcel  : Expecting binder but got null!
com.termux.x11 logcat: 03-04 14:34:36.437 30227 30227 D ViewRootImpl@8fce065[MainActivity]: onSyncComplete
com.termux.x11 logcat: 03-04 14:34:36.437 30227 30227 D ViewRootImpl@8fce065[MainActivity]: setupSync seqId=0 mSyncId=0 fn=1 caller=android.view.ViewRootImpl$$ExternalSyntheticLambda11.accept:6 android.window.SurfaceSyncer.lambda$setupSync$1$android-window-SurfaceSyncer:128 android.window.SurfaceSyncer$$ExternalSyntheticLambda1.accept:8 android.window.SurfaceSyncer$SyncSet.checkIfSyncIsComplete:382 android.window.SurfaceSyncer$SyncSet.markSyncReady:359 android.window.SurfaceSyncer.markSyncReady:151 android.view.ViewRootImpl.performTraversals:4373
com.termux.x11 logcat: 03-04 14:34:36.437 30227 30227 D ViewRootImpl@8fce065[MainActivity]: reportDrawFinished seqId=0 mSyncId=-1 fn=1 mSurfaceChangedTransaction=0x6e8d285b30
com.termux.x11 logcat: 03-04 14:34:36.440 30227 30227 D InsetsController: onStateChanged: InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(468, 0 - 612, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={2.8125} cutoutSpec={M 0,0 H -25.6 V 28.44444444444444 H 25.6 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=96, center=Point(96, 96)}, RoundedCorner{position=TopRight, radius=96, center=Point(984, 96)}, RoundedCorner{position=BottomRight, radius=96, center=Point(984, 2304)}, RoundedCorner{position=BottomLeft, radius=96, center=Point(96, 2304)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mSources= { InsetsSource: {mType=ITYPE_STATUS_BAR, mFrame=[0,0][1080,80], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_NAVIGATION_BAR, mFrame=[0,2256][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_GESTURES, mFrame=[0,0][0,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_GESTURES, mFrame=[1080,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_MANDATORY_GESTURES, mFrame=[0,0][1080,116], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_MANDATORY_GESTURES, mFrame=[0,2256][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_DISPLAY_CUTOUT, mFrame=[0,0][-100000,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_DISPLAY_CUTOUT, mFrame=[0,0][1080,80], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_DISPLAY_CUTOUT, mFrame=[100000,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_DISPLAY_CUTOUT, mFrame=[0,100000][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_TAPPABLE_ELEMENT, mFrame=[0,0][1080,80], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_TAPPABLE_ELEMENT, mFrame=[0,2256][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_IME, mFrame=[0,0][0,0], mVisibleFrame=[0,1374][1080,2400], mVisible=false, mInsetsRoundedCornerFrame=false} } host=com.termux.x11/com.termux.x11.MainActivity from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:6534
com.termux.x11 logcat: 03-04 14:34:36.442 30227 30227 D InsetsController: controlAnimationUnchecked: Added types=3 animType=1 host=com.termux.x11/com.termux.x11.MainActivity from=android.view.InsetsController.applyAnimation:1576 android.view.InsetsController.applyAnimation:1557 android.view.InsetsController.onControlsChanged:995
com.termux.x11 logcat: 03-04 14:34:36.442 30227 30227 D InsetsController: onStateChanged: InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(468, 0 - 612, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={2.8125} cutoutSpec={M 0,0 H -25.6 V 28.44444444444444 H 25.6 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=96, center=Point(96, 96)}, RoundedCorner{position=TopRight, radius=96, center=Point(984, 96)}, RoundedCorner{position=BottomRight, radius=96, center=Point(984, 2304)}, RoundedCorner{position=BottomLeft, radius=96, center=Point(96, 2304)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mSources= { InsetsSource: {mType=ITYPE_STATUS_BAR, mFrame=[0,0][1080,80], mVisible=false, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_NAVIGATION_BAR, mFrame=[0,2256][1080,2400], mVisible=false, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_GESTURES, mFrame=[0,0][0,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_GESTURES, mFrame=[1080,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_MANDATORY_GESTURES, mFrame=[0,0][1080,116], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_MANDATORY_GESTURES, mFrame=[0,2256][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_DISPLAY_CUTOUT, mFrame=[0,0][-100000,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_DISPLAY_CUTOUT, mFrame=[0,0][1080,80], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_DISPLAY_CUTOUT, mFrame=[100000,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_DISPLAY_CUTOUT, mFrame=[0,100000][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_TAPPABLE_ELEMENT, mFrame=[0,0][1080,80], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_TAPPABLE_ELEMENT, mFrame=[0,2256][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_IME, mFrame=[0,0][0,0], mVisibleFrame=[0,1374][1080,2400], mVisible=false, mInsetsRoundedCornerFrame=false} } host=com.termux.x11/com.termux.x11.MainActivity from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:6521
com.termux.x11 logcat: 03-04 14:34:36.443 30227 30227 I ViewRootImpl@8fce065[MainActivity]: handleResized, msg = 5 frames=ClientWindowFrames{frame=[0,80][1080,2400] display=[0,80][1080,2400] parentFrame=[0,0][0,0] parentClippedByDisplayCutout=false} forceNextWindowRelayout=false displayId=0 resizeMode=-1 frameChanged=false displayFrameChanged=false configChanged=false displayChanged=false
com.termux.x11 logcat: 03-04 14:34:36.443 30227 30227 I ViewRootImpl@8fce065[MainActivity]: handleResized mSyncSeqId = 0
com.termux.x11 logcat: 03-04 14:34:36.443 30227 30227 D ViewRootImpl@8fce065[MainActivity]: reportNextDraw android.view.ViewRootImpl.handleResized:2380 android.view.ViewRootImpl.-$$Nest$mhandleResized:0 android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:6522 android.view.ViewRootImpl$ViewRootHandler.handleMessage:6491 android.os.Handler.dispatchMessage:106
com.termux.x11 logcat: 03-04 14:34:36.443 30227 30227 I SurfaceView@7dd0040: finishedDrawing
com.termux.x11 logcat: 03-04 14:34:36.448 30227 30227 D ViewRootImpl@8fce065[MainActivity]: Setup new sync id=1
com.termux.x11 logcat: 03-04 14:34:36.448 30227 30227 D ViewRootImpl@8fce065[MainActivity]: Setting syncFrameCallback
com.termux.x11 logcat: 03-04 14:34:36.448 30227 30227 E SurfaceSyncer: Failed to find sync for id=0
com.termux.x11 logcat: 03-04 14:34:36.448 30227 30227 D ViewRootImpl@8fce065[MainActivity]: registerCallbacksForSync syncBuffer=false
com.termux.x11 logcat: 03-04 14:34:36.461 30227 30275 D ViewRootImpl@8fce065[MainActivity]: Received frameDrawingCallback syncResult=0 frameNum=2.
com.termux.x11 logcat: 03-04 14:34:36.462 30227 30275 D ViewRootImpl@8fce065[MainActivity]: Setting up sync and frameCommitCallback
com.termux.x11 logcat: 03-04 14:34:36.464 30227 30253 D ViewRootImpl@8fce065[MainActivity]: Received frameCommittedCallback lastAttemptedDrawFrameNum=2 didProduceBuffer=true
com.termux.x11 logcat: 03-04 14:34:36.464 30227 30227 D ViewRootImpl@8fce065[MainActivity]: onSyncComplete
com.termux.x11 logcat: 03-04 14:34:36.465 30227 30227 D ViewRootImpl@8fce065[MainActivity]: setupSync seqId=0 mSyncId=1 fn=2 caller=android.view.ViewRootImpl$$ExternalSyntheticLambda11.accept:6 android.window.SurfaceSyncer.lambda$setupSync$1$android-window-SurfaceSyncer:128 android.window.SurfaceSyncer$$ExternalSyntheticLambda1.accept:8 android.window.SurfaceSyncer$SyncSet.checkIfSyncIsComplete:382 android.window.SurfaceSyncer$SyncSet.markSyncReady:359 android.window.SurfaceSyncer.markSyncReady:151 android.view.ViewRootImpl.performTraversals:4373
com.termux.x11 logcat: 03-04 14:34:36.465 30227 30227 D ViewRootImpl@8fce065[MainActivity]: reportDrawFinished seqId=0 mSyncId=-1 fn=2 mSurfaceChangedTransaction=0x6e8d285b30
com.termux.x11 logcat: 03-04 14:34:36.469 30227 30227 I ViewRootImpl@8fce065[MainActivity]: MSG_WINDOW_FOCUS_CHANGED 1 0
com.termux.x11 logcat: 03-04 14:34:36.469 30227 30227 D ViewRootImpl@8fce065[MainActivity]: mThreadedRenderer.initializeIfNeeded()#2 mSurface={isValid=true 0x6f5d2b19d0}
com.termux.x11 logcat: 03-04 14:34:36.470 30227 30227 I DecorView: notifyKeepScreenOnChanged: keepScreenOn=true
com.termux.x11 logcat: 03-04 14:34:36.470 30227 30227 I DecorView: notifyKeepScreenOnChanged: keepScreenOn=true
com.termux.x11 logcat: 03-04 14:34:36.471 30227 30227 D InputMethodManager: startInputInner - Id : 0
com.termux.x11 logcat: 03-04 14:34:36.471 30227 30227 I InputMethodManager: startInputInner - mService.startInputOrWindowGainedFocus
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]: performTraversals params={(0,0)(fillxfill) sim={adjust=resize} ty=BASE_APPLICATION wanim=0x1030309
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fl=81810580
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   pfl=1a020040
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   vsysui=1706
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   bhv=SHOW_TRANSIENT_BARS_BY_SWIPE
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fitSides= naviIconColor=0
com.termux.x11 logcat: 03-04 14:34:36.476 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   sfl=100000}
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]: performTraversals mFirst=false windowShouldResize=false viewVisibilityChanged=false mForceNextWindowRelayout=false params={(0,0)(fillxfill) sim={adjust=resize} ty=BASE_APPLICATION fmt=TRANSLUCENT wanim=0x1030309
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fl=81810580
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   pfl=1a020040
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   vsysui=1706
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   bhv=SHOW_TRANSIENT_BARS_BY_SWIPE
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   fitSides= naviIconColor=0
com.termux.x11 logcat: 03-04 14:34:36.477 30227 30227 D ViewRootImpl@8fce065[MainActivity]:   sfl=100000}
com.termux.x11 logcat: 03-04 14:34:36.493 30227 30227 D BLASTBufferQueue: update, w= 1080 h= 2320 mNativeObject= 0x6ecd28bc30 sc.mNativeObject= 0x6e5d2b9150 format= -3 caller= android.view.ViewRootImpl.updateBlastSurfaceIfNeeded:2778 android.view.ViewRootImpl.relayoutWindow:9635 android.view.ViewRootImpl.performTraversals:3759 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665 android.view.Choreographer$CallbackRecord.run:1301
com.termux.x11 logcat: 03-04 14:34:36.493 30227 30227 I ViewRootImpl@8fce065[MainActivity]: Relayout returned: old=(0,80,1080,2400) new=(0,80,1080,2400) req=(1080,2320)0 dur=16 res=0x0 s={true 0x6f5d2b19d0} ch=false seqId=0
com.termux.x11 logcat: 03-04 14:34:36.494 30227 30227 I ViewRootImpl@8fce065[MainActivity]: updateBoundsLayer: t = android.view.SurfaceControl$Transaction@16ae313 sc = Surface(name=Bounds for - com.termux.x11/com.termux.x11.MainActivity@0)/@0xf64a950 frame = 4
com.termux.x11 logcat: 03-04 14:34:36.494 30227 30227 I ViewRootImpl@8fce065[MainActivity]: mWNT: t=0x6e8d288950 mBlastBufferQueue=0x6ecd28bc30 fn= 4 caller= android.view.ViewRootImpl.prepareSurfaces:2865 android.view.ViewRootImpl.performTraversals:4108 android.view.ViewRootImpl.doTraversal:2991
com.termux.x11 logcat: 03-04 14:34:36.494 30227 30227 D BLASTBufferQueue: mergeWithNextTransaction, mNativeObject= 0x6ecd28bc30 t.mNativeObject= 0x6e8d288950 frameNumber= 4 caller= android.view.ViewRootImpl.mergeWithNextTransaction:12594 android.view.ViewRootImpl.prepareSurfaces:2865 android.view.ViewRootImpl.performTraversals:4108 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665 android.view.Choreographer$CallbackRecord.run:1301
com.termux.x11 logcat: 03-04 14:34:36.496 30227 30227 D InputMethodManager: startInputInner - Id : 0
com.termux.x11 logcat: 03-04 14:34:36.797 30227 30227 D InsetsController: cancelAnimation of types: 3, animType: 1, host: com.termux.x11/com.termux.x11.MainActivity, from:android.view.InsetsController.notifyFinished:1375 android.view.InsetsAnimationThreadControlRunner$1.lambda$notifyFinished$0$android-view-InsetsAnimationThreadControlRunner$1:84 android.view.InsetsAnimationThreadControlRunner$1$$ExternalSyntheticLambda1.run:4
```
</details>

**Comment by twaik** [comment]
Do you use proot or something?

**Comment by RalfWerner** [comment]
No! only ssh to start on PC with real Keyboard.

**Comment by twaik** [comment]
Can I connect to you to check what a problem can it be?

**Comment by RalfWerner** [comment]
by TeamViewer on my PC(1079799935)?

**Comment by twaik** [comment]
I need a password too.

**Comment by twaik** [comment]
But not here

**Comment by twaik** [comment]
Send me in https://twaik.t.me/

**Comment by RalfWerner** [comment]
Sorry I've removed all my Telegram-Installation from my PC, but have TeamViewer QuickSupport on my new phone (1 517 112 947) with confirm.

**Comment by twaik** [comment]
Does not work in my case. Try [this](https://remotedesktop.google.com/support).

**Comment by RalfWerner** [comment]
My code: 737909813718

**Comment by RalfWerner** [comment]
Thanks for Support. Xwayland package was'nt updated!
I've to study the behavior of this *new* X11 Version in relation to the old once and VNC. but not today.

**Comment by twaik** [comment]
@RalfWerner Now you can change extra keys list in settings.

**Comment by RalfWerner** [comment]
Thank you @twaik I try your two sources (from your artifact - behind: `[~]$ ll ...`), when you **confirm - is that Ok?**
The last problem (commit [above](#issuecomment-1454743529) edited) had to do with `pkg up` so I did it again and got the following log:
```
[~]$ ll *deb *apk
-rw-rw---- 1 u0_a259 media_rw 15789760 Mar  9 07:58 app-debug.apk
-rw-rw---- 1 u0_a259 media_rw     5538 Mar  9 07:58 termux-x11-1.02.07-0-all.deb
[~]$ dpkg -i termux-x11-1.02.07-0-all.deb
(Reading database ... 41910 files and directories currently installed.)
Preparing to unpack termux-x11-1.02.07-0-all.deb ...
Unpacking termux-x11 (1.02.07-0) over (1.02.07) ...
Setting up termux-x11 (1.02.07-0) ...
[~]$ pkg up
...
[*] (1) http://mirror.mephi.ru/termux/termux-main: ok
Picking mirror: (0) /data/data/com.termux/files/usr/etc/termux/mirrors/russia/mirror.mephi.ru
Get:1 http://mirror.mephi.ru/termux/termux-main stable InRelease [14.0 kB]
Hit:2 https://its-pointless.github.io/files/24 termux InRelease
Ign:1 http://mirror.mephi.ru/termux/termux-main stable InRelease
Get:3 http://mirror.mephi.ru/termux/termux-x11 x11 InRelease [14.0 kB]
Ign:3 http://mirror.mephi.ru/termux/termux-x11 x11 InRelease
Get:1 http://mirror.mephi.ru/termux/termux-main stable InRelease [14.0 kB]
Get:3 http://mirror.mephi.ru/termux/termux-x11 x11 InRelease [14.0 kB]
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [491 kB]
Ign:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages
Get:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages [123 kB]
Ign:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [491 kB]
Ign:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages
Get:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages [123 kB]
Ign:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [491 kB]
Ign:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages
Get:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages [123 kB]
Ign:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [491 kB]
Ign:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages
Get:5 http://mirror.mephi.ru/termux/termux-x11 x11/main aarch64 Packages [123 kB]
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [491 kB]
Err:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages
  Connection timed out [IP: 85.143.112.112 80]
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [613 kB]
Err:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages
  Error reading from server. Remote end closed connection [IP: 85.143.112.112 80]
Get:4 http://mirror.mephi.ru/termux/termux-main stable/main aarch64 Packages [1566 kB]
Fetched 1583 kB in 11min 6s (2377 B/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
69 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following NEW packages will be installed:
  abseil-cpp libssh python-ensurepip-wheels
The following packages will be upgraded:
  aapt android-tools apt clang command-not-found curl dbus dialog dos2unix dpkg ffmpeg fontconfig fontconfig-utils freetype git glib glib-bin grep
  gtk-update-icon-cache gtk3 harfbuzz imagemagick imath iproute2 less libaom libass libcompiler-rt libcurl libdav1d libdrm libgnutls libhandy
  libheif libjpeg-turbo libllvm libnghttp2 libnss libprotobuf libsqlite libvpx libwayland libwayland-protocols libx11 libxml2 littlecms lld llvm
  make mesa mpv-x nss-utils opencv p7zip pango proot pulseaudio python python-pip tar termux-tools tigervnc unrar xorg-font-util xorg-fonts-75dpi
  xorg-fonts-alias xorg-fonts-encodings xorg-util-macros zstd
69 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 165 MB of archives.
After this operation, 12.4 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```
The only mirror that is currently available, come from **Russia is that trustworthy?**

**Comment by twaik** [comment]
1. `apt install termux-x11-1.02.07-0-all.deb`
2. Companion package can be kept it is made to be universal.
3. All the mirrors are controlled by termux so I think so.

**Comment by twaik** [comment]
Termux:X11 is a separate application and has separate settings. Open preferences, the last line.

**Comment by RalfWerner** [comment]
I've just closed EK lines (three fingers down), found no notification to change preferences and xfonds are very small in relation to previous X11 versions. **Is there a change description** for _**old X11 users**_ to handle the new environment?
**Could I use the same > EK=.termux/termux.properties like above?**
XClients here in landscape: fluxbox, xterm, pcmanfm (see > Support by TeamViewer [above](#issuecomment-1454743529))
![grafik](https://user-images.githubusercontent.com/45426380/223991760-91af2164-e347-40b7-92d9-3bee48edba26.png)
X11-basic use in fullscreen mode the Navbar+Top-Line as X11 Space - the new one not!

**Comment by twaik** [comment]
Do you have Android 13? Posting notification is restricted for an app in that case, you should give it that permission manually, through Android settings.

There is no way to use config files. But you can simply copy `extra-keys` value to new app, it should work.

About fonts: older versioms used device's DPI value, but it was a problem on some environments so I changed it to be 96 by default.  I'll add option to change it.

**Comment by RalfWerner** [comment]
> Do you have **Android 13**? Posting **notification** is restricted for an app in that case, you should give it that permission.

**Yes 13** on this device (A33) **notification** (Benachrichtig..) are set. But still **no way to Preferences** or **restore EK**!
![grafik](https://user-images.githubusercontent.com/45426380/224001356-b1f959d1-0d91-4098-b7a2-11591398c5ff.png)
> About fonts: older versioms used device's DPI value, but it was a problem on some environments so I changed it to be 96 by default. I'll add an option to change it.

What is is to **do**, to get the **same font as before** (here the same XClients on HUAWEI with **real** fullscreen of X11 basic)? 
![grafik](https://user-images.githubusercontent.com/45426380/224007378-59794a84-6938-4796-b43e-3939eed82a03.png)

**Comment by twaik** [comment]
Actually there is an option to specify DPI using `termux-x11 -dpi 200` or something that like.

**Comment by twaik** [comment]
<details>

https://user-images.githubusercontent.com/9674930/224071263-ba665fba-6094-4a92-88ab-38cae60f348d.mp4
</details>

Did you paste the whole config? You should use there only `extra-keys` value.

**Comment by twaik** [comment]
Where exactly did you pass `-dpi` option? And how exactly?

**Comment by RalfWerner** [comment]
Nice - Preferences with long tab:) and **restore EK?** to `-dpi` like [above](#issuecomment-1454735988) but `i=200; termux-x11 -dpi $i :0&`.
To **EK**: Only the _one line_ inclusive `extra-keys=` (you can repeat this).

**Comment by twaik** [comment]
> To EK: Only the one line inclusive extra-keys= (you can repeat this).

Use `\` symbol at end of line, or use it single line.

> to -dpi like [above](https://github.com/termux/termux-x11/issues/54#issuecomment-1454735988) but i=200; termux-x11 -dpi $i :0&.

**Comment by RalfWerner** [comment]
![grafik](https://user-images.githubusercontent.com/45426380/224081654-f2797b4e-5670-4829-9448-a0a6e1729698.png)
Shots ti extra-keys. Extend by `/` has no effect:(

**Comment by twaik** [comment]
Wrong slash.

**Comment by RalfWerner** [comment]
Same with `\` - what do you get with [cut/paste](#issuecomment-1462239083)?

**Comment by twaik** [comment]
> `[[{macro:CTRL,display:'^',popup:{macro:'CTRL d',display:d}},{key:TAB,popup:PASTE},{key:LEFT,popup:BKSP},{key:RIGHT,popup:{macro:'o ENTER',display:off}},{key:UP,popup:PGUP},  {key:DOWN,popup:PGDN},{key:END,popup:HOME},{key:ENTER,popup:{macro:'CTRL n',display:'next'}},{key:KEYBOARD,popup:DRAWER},{macro:'mc SPACE i ENTER',display:mc,popup:F10},{display:X11,macro:'t SPACE 0 ENTER',popup:{macro:'w ENTER',display:WM}}]]`

It seems to be single line. Use this as is.

**Comment by RalfWerner** [comment]
Ok - it works without any slash and keyword:`extra-keys=`:) Typically, the _properties_ are created by a script. Could I also write/copy this in the X11 storage if a _symlink_ does not work? the **EK** `MC` in **xterm** works well inlusive _popup_ `F10` to open/close **mc** and all cursor movements. KB icon (toggle) and popup `Drawer` (preferences) also works.
But I cant read anything (too small) and KB-toggle destroy WM-layout

**Comment by twaik** [comment]
Try to change resolution to something that fits better. I am already working on DPI option code.

**Comment by RalfWerner** [comment]
Text and window dimensions are only preserved at **native**. _scaled+(30-200)%_ and _exact+resultion, even with 200% _scaled_, hardly read anywhere except for a very small _resultion_, which do **not change** with the rotation of the device or pop view (PiP). My opinion: nice game without benefit.

I still haven't found a way to _**reactivate**_ the **EK** when _off_ (except restart) and 3 fingers down is a typical gesture from _direct mode_ when it is available again. But  **great thanks** for your fantastic work until here. For me is the end of work today.

**Comment by naydana890** [comment]
Zip file dowload please

**Comment by RalfWerner** [comment]
> Zip file dowload please

... use [my](https://www.dropbox.com/s/ug071qoox8gwf1c/ip.zip?dl=0) copies (dropbox) above?
In between I check 5 Version [Setup shots on five devices](https://github.com/termux/termux-x11/assets/45426380/638c958b-6f2d-42ce-8c9f-d66bee7aea1f) the newest are in [Actions](https://github.com/termux/termux-x11/actions)

**Comment by RalfWerner** [comment]
To [EK](https://github.com/termux/termux-app/issues/3601#issuecomment-1750466959) in `Termux-app` and `-x11` in Preparation (Edit) have patience.
>**Did not understand that** (EK is currently (x11a) completely overwriting my task manager).

>**What??** (shows only part of the EK as symbols)

>That is an old version which is very outdated...

>It is done for purpose to let users work with EK without changing virtual display resolution
